Makes the faction variable into a list

Tomeno opened the pull request:
This change allows for multiple factions, making mapper lifes easier.
wb(wild-billy) made the CanAttack() proc.

RemieRichards commented on NTstation13.int:
```diff
@@ -1,6 +1,11 @@
 // BEGIN_INTERNALS
 /*
 MAP_ICON_TYPE: 0
+WINDOW: code\game\gamemodes\antag_spawner.dm;code\game\gamemodes\blob\overmind.dm;code\game\gamemodes\blob\blobs\factory.dm;code\game\gamemodes\nuclear\nuclear.dm;code\modules\mob\living\silicon\robot\robot.dm;code\modules\mob\living\simple_animal\hostile\alien.dm;code\modules\mob\living\simple_animal\hostile\bear.dm;code\modules\mob\living\simple_animal\hostile\carp.dm;code\modules\mob\living\si ...
```
Revert the changes to this file.
This file should not be changed.

Tomeno committed:
Makes the faction variable into a list

This change allows for multiple factions, making mapper lifes easier.
wb(wild-billy) made the CanAttack() proc.
Now with 20% less NTstation.int!

WJohn commented:
:+1:
This will be a lot of fun for mappers to play with! Thank you <3
